File: solo-leveling-player-model/src/main/java/com/sleepkqq/sololeveling/player/model/entity/user/enums/UserRole.java
package com.sleepkqq.sololeveling.player.model.entity.user.enums;

import com.sleepkqq.sololeveling.jimmer.enums.EnumPathGenerator;
import com.sleepkqq.sololeveling.jimmer.enums.LocalizableEnum;
import lombok.Getter;
import org.babyfish.jimmer.sql.EnumItem;

@Getter
public enum UserRole implements LocalizableEnum {
  @EnumItem(ordinal = 0)
  USER,

  @EnumItem(ordinal = 1)
  ADMIN,

  @EnumItem(ordinal = 2)
  DEVELOPER,

  @EnumItem(ordinal = 3)
  MANAGER;

  private final String path = EnumPathGenerator.generatePath(this);
}

File: solo-leveling-player-model/src/main/kotlin/com/sleepkqq/sololeveling/player/model/entity/user/enums/UserRole.kt
package com.sleepkqq.sololeveling.player.model.entity.user.enums

import org.babyfish.jimmer.sql.EnumItem

enum class UserRole {
	@EnumItem(ordinal = 0)
	USER,

	@EnumItem(ordinal = 1)
	ADMIN
}